for_developers/product_design.md

```diff
@@ -0,0 +1,114 @@
+
+## Product Design Principles
+
+To realize our product missions, we establish the following design principles and details, providing a comprehensive overview of our product's structure.
+
+### Core
+
+1. **Static Typing System for Task-Data-Model to Accelerate Task and Model Development Cycles**
+
+   _"Advocates of static typing argue that ... a better design time developer experience..."_[^1]
+
+   Our framework should support various deep learning tasks simultaneously.
+   To achieve this mission, it is essential to make it developer-friendly for expanding our product in both task and model dimensions.
+   We believe that nice abstractions with static typing can help developers in this regard.
+   We introduced the **Task-Data-Model** abstraction with static typing.
+
+   | ![Task-Data-Model](./images/product_design/task_data_model.png) |
+   | :-------------------------------------------------------------: |
+   |    Figure 1. Task-Data-Model abstractions with static typing    |
+
+   As shown in Figure 1, there are abstractions (grey-colored boxes) that form the base for each task:
+
+   1. `OTXTaskType`: A Python enum defining the deep learning task supported in this framework.
+   2. `OTXDataEntity`: A Python dataclass object representing a single data sample for each `OTXTaskType`.
+   3. `OTXBatchDataEntity`: A Python dataclass object representing a batch of `OTXDataEntity`.
+   4. `OTXModel`: PyTorch module (`nn.Module`) defined for each `OTXTaskType`.
+      Its `forward()` function has strict typing: 1) `OTXBatchDataEntity` as inputs and 2) `OTXBatchLossEntity` or `OTXBatchPredEntity` as outputs.
+
+   On the other hand, we can see embodiments from the abstractions (blue-colored boxes) for the detection task (`OTXTaskType.DETECTION`):
+
+   1. `DetDataEntity` and `DetBatchDataEntity`: Python dataclass objects including `bboxes` and `labels` fields for the detection task.
+      `DetBatchDataEntity` is used for the input of `OTXDetectionModel`'s `forward()` function.
+   2. `DetBatchPredEntity`: Python dataclass object including the `scores` field beyond `DetBatchDataEntity`.
+      It is used for the output of `OTXDetectionModel`'s `forward()` function at inference mode.
+
+   As observed, a developer aiming to add a new model to OTX for the detection task only needs to consider the strictly typed input and output (`DetBatchDataEntity` and `DetBatchPredEntity`).
+
+   Conversely, in the event of a new task requirement, we can incorporate it by implementing embodiments from the base, such as `OTXTaskType`, `OTXDataEntity`, `OTXBatchDataEntity`, and `OTXModel`.
+
+2. **Single Engine but Capable of Incorporating Multiple Popular Model Training Frameworks**
+
+   _"Don't Reinvent the Wheel"_[^2]
+
+   We chose PyTorch Lightning[^3] as the primary engine for our training framework.
```

Comment:
`chose` -> `choose`. I think it is better to keep using same verb tense in overall document.

Reply:
```suggestion
   We choose PyTorch Lightning[^3] as the primary engine for our training framework.
```